src/platform/sdl/main.c

```diff
@@ -321,3 +337,64 @@ static void mSDLDeinit(struct mSDLRenderer* renderer) {
 
 	SDL_Quit();
 }
+
+void setLogger(struct mCore* core) {
+	int fakeBool = 0;
+	bool logToStdout = true;
+	bool logToFile = false;
+
+	if (mCoreConfigGetIntValue(&core->config, "logToStdout", &fakeBool))
+		logToStdout = fakeBool;
+	if (mCoreConfigGetIntValue(&core->config, "logToFile", &fakeBool))
+		logToFile = fakeBool;
+	const char* logFile = mCoreConfigGetValue(&core->config, "logFile");
+	int filterLevels = core->opts.logLevel;
+
+	_setLogger(logToStdout, logToFile, logFile, filterLevels);
+}
+
+void _setLogger(bool logToStdout, bool logToFile, const char* logFile, int filterLevels) {
+	// Assign basic static variables
+	_logToStdout = logToStdout;
+	_logFile = NULL;
+	
+	if(logToFile && logFile)
+		_logFile = VFileOpen(logFile, O_WRONLY | O_CREAT | O_TRUNC);
```

Comment:
I don't believe the Qt frontend truncates the log file; I believe it appends instead. It's best to keep this consistent.

Reply:
Fixed in https://github.com/mgba-emu/mgba/pull/2345/commits/9440ca24e9b66396445955640548bab981801c2d .